torch/jit/__init__.py

```diff
@@ -1711,7 +1711,11 @@ def _convert_to_script_module(mod, methods=None):
 
     def make_stub(method):
         func = get_function_from_type(type(mod), method)
-        return script_method(func, createResolutionCallbackFromClosure(func))
+        try:
+            sm = script_method(func, createResolutionCallbackFromClosure(func))
+        except Exception as e:
+            raise Exception(f"Error scripting method {func.__name__} in module {mod.__class__.__name__}: {str(e)}")
```

Comment:
Aren't file / line numbers already included in compilation errors?

Reply:
doesn't look like it always is.

```
    def test_errors_1(self):
        class N(torch.nn.Module):
            def __init__(self):
                super(N, self).__init__()

        class M(torch.nn.Module):
            def __init__(self):
                super(M, self).__init__()
                self.n = N()

            def forward(self, a):
                self.n(a)
                return a

        self.checkModule(M(), (torch.ones(1),))
```

error before:
```
ERROR: test_errors_1 (test_jit.TestRecursiveScript)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/users/bwasti/pytorch/test/test_jit.py", line 13025, in test_errors_1
    self.checkModule(M(), (torch.ones(1),))
  File "/data/users/bwasti/pytorch/test/test_jit.py", line 12958, in checkModule
    sm = torch.jit.script(nn_module)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1068, in script
    return _convert_to_script_module(obj)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1722, in _convert_to_script_module
    return WeakScriptModuleProxy(mod, stubs)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1300, in init_then_register
    original_init(self, *args, **kwargs)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1635, in __init__
    _create_methods_from_stubs(self, stubs)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1261, in _create_methods_from_stubs
    self._c._create_methods(self, defs, rcbs, defaults)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 968, in _make_strong_submodule
    new_strong_submodule = _convert_to_script_module(module)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1721, in _convert_to_script_module
    stubs = list(map(make_stub, methods))
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1717, in make_stub
    raise e
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1715, in make_stub
    sm = script_method(func, createResolutionCallbackFromClosure(func))
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1106, in script_method
    ast = get_jit_def(fn, self_name="ScriptModule")
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/frontend.py", line 169, in get_jit_def
    return build_def(ctx, py_ast.body[0], type_line, self_name)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/frontend.py", line 198, in build_def
    param_list = build_param_list(ctx, py_def.args, self_name)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/frontend.py", line 218, in build_param_list
    raise ValueError(_vararg_kwarg_err)
ValueError: Compiled functions can't take variable number of arguments or use keyword-only arguments with defaults
```

error after:
```
======================================================================
ERROR: test_errors_1 (test_jit.TestRecursiveScript)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1715, in make_stub
    sm = script_method(func, createResolutionCallbackFromClosure(func))
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1106, in script_method
    ast = get_jit_def(fn, self_name="ScriptModule")
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/frontend.py", line 169, in get_jit_def
    return build_def(ctx, py_ast.body[0], type_line, self_name)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/frontend.py", line 198, in build_def
    param_list = build_param_list(ctx, py_def.args, self_name)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/frontend.py", line 218, in build_param_list
    raise ValueError(_vararg_kwarg_err)
ValueError: Compiled functions can't take variable number of arguments or use keyword-only arguments with defaults

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/data/users/bwasti/pytorch/test/test_jit.py", line 13025, in test_errors_1
    self.checkModule(M(), (torch.ones(1),))
  File "/data/users/bwasti/pytorch/test/test_jit.py", line 12958, in checkModule
    sm = torch.jit.script(nn_module)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1068, in script
    return _convert_to_script_module(obj)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1721, in _convert_to_script_module
    return WeakScriptModuleProxy(mod, stubs)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1300, in init_then_register
    original_init(self, *args, **kwargs)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1635, in __init__
    _create_methods_from_stubs(self, stubs)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1261, in _create_methods_from_stubs
    self._c._create_methods(self, defs, rcbs, defaults)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 968, in _make_strong_submodule
    new_strong_submodule = _convert_to_script_module(module)
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1720, in _convert_to_script_module
    stubs = list(map(make_stub, methods))
  File "/data/users/bwasti/python3_dev/lib64/python3.6/site-packages/torch/jit/__init__.py", line 1717, in make_stub
    raise Exception(f"Error scripting method {func.__name__} in module {mod.__class__.__name__}: {str(e)}")
Exception: Error scripting method forward in module N: Compiled functions can't take variable number of arguments or use keyword-only arguments with defaults
```